Homework/2/Program.cs

```diff
@@ -0,0 +1,390 @@
+
+            if (CheckIfUserExists(username))
+            {
+                Console.WriteLine("userexists");
+                string[] userDetails = ReturnUserByUsername(username);
+                Console.WriteLine(userDetails[0] + " " + userDetails[1]);
+                if (userDetails[1] == password) {
+                    loggedIn = true;
+                    
+                    MenuSelection();
+                    return;
+                }
+            }
+
+            if (RetryConfirmation("Bad username or password."))
+            {
+                Login();
+            }
+            else {
+                MenuSelection();
+            }
+        }
+
+        static void LogOut() {
+            Console.Clear();
+            loggedIn = false;
+            MenuSelection();
+        }
+
+        // Function to display error message and return true for retry;
+        static bool RetryConfirmation(string messageToDisplay) {
+            Console.Clear();
+            Console.WriteLine(messageToDisplay+" Retry? y/n");
+
+            if (Console.ReadLine().ToLower() == "y")
+            {
+                return true;
+            }
+            return false;
+        }
+
+
+        // Returns true if username is found
+        static bool CheckIfUserExists(string username) {
+            for (int i = 0; i < userArray.Length; i++)
+            {
+                if (userArray[i][0].ToLower() == username.ToLower())
+                {
+                    Console.WriteLine("Found");
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        // Look through user array for a username and return string array of username and password
+        static string[] ReturnUserByUsername(string username) {
+            string[] array = new string[] {"",""};
+            for (int i = 0; i < userArray.Length; i++)
+            {
+                if (userArray[i][0].ToLower() == username.ToLower())
+                {
+                    array = userArray[i];
+                }
+            }
+            return array;
+        }
+
+        //Adding string[] - [username,password] - user details to userArray at selected index.
+        static void AddUserToArrayAtIndex(string[] userDetails, int index) {
+            string[][] newArray = new string[userArray.Length + 1][];
+
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                if (index == 0 && i == 0 )
+                {
+                    newArray[i] = userDetails;
+                    continue;
+                }
+
+                if (i < index)
+                {
+                    newArray[i] = userArray[i];
+                }
+
+                if (i==index)
+                {
+                    newArray[i] = userDetails;
+                }
+
+                if (i > index)
+                {
+                    newArray[i] = userArray[i - 1];
+                }
```

Comment:
Only one of these conditions shuold be true at a time.
Therefore, use `else if` and not `if`. This will also allow you to remove the continue statement, because the if that is hit will stop checking others further ifs.